### PR TITLE
Detect RV300 and VC4 hardware and disable slow half float vertex on them

### PR DESCRIPTION
**Pull request by illwieckz**
Import from:

- https://github.com/DaemonEngine/Daemon/pull/1179

### CODE REVIEW

**Comment by illwieckz** [comment]
I also added VC4 detection, it's twice faster with float vertex on it.

Before (25fps):

unvanquished_2024-10-11_183704_000

After (55fps):

unvanquished_2024-10-11_184004_000

File: src/engine/sys/sdl_glimp.cpp [hunk 1]
```diff
@@ -2210,10 +2214,65 @@ static void GLimp_InitExtensions()
...
+	instead. */
+	{
+		bool halfFloatVertexEnabled = r_arb_half_float_vertex.Get();
+
+		if ( glConfig2.driverVendor == glDriverVendor_t::MESA
+			&& glConfig2.hardwareVendor == glHardwareVendor_t::ATI )
+		{
+			bool foundRv300 = false;
+
+			std::string cardName = "unknown ATI RV3xx";
```
Comment (VReaperV) [inline src/engine/sys/sdl_glimp.cpp:2231]: The initial value is unused.

Comment (illwieckz) [reply src/engine/sys/sdl_glimp.cpp:2231]: It is used if I forgot to list a card name and the `glConfig.hardwareType == glHardwareType_t::GLHW_R300 && glConfig2.maxAluInstructions == 64` test matches.

Comment (slipher) [reply src/engine/sys/sdl_glimp.cpp:2231]: It isn't. It's assigned the first time the for loop is entered.

File: src/engine/sys/sdl_glimp.cpp [hunk 2]
```diff
@@ -2210,10 +2214,65 @@ static void GLimp_InitExtensions()
...
+			}
+		}
+
+		// VAO and VBO
+		// made required in OpenGL 3.0
+		glConfig2.halfFloatVertexAvailable = LOAD_EXTENSION_WITH_TEST( ExtFlag_CORE, ARB_half_float_vertex, halfFloatVertexEnabled );
+
+		if ( !halfFloatVertexEnabled )
+		{
+			logger.Warn( "Missing half-float vertex, using float vertex instead." );
```
Comment (slipher) [inline src/engine/sys/sdl_glimp.cpp:2273]: How about Notice for this one. We shouldn't use Warn so much unless there is a serious problem.

Comment (illwieckz) [reply src/engine/sys/sdl_glimp.cpp:2273]: I like to have a bolder log for such things, especially when it is reporting about making decisions, it makes it easier to catch them in logs. Though I agree Warn is maybe too much.

File: src/engine/sys/sdl_glimp.cpp (hunk 1 above)
Comment (illwieckz) [reply src/engine/sys/sdl_glimp.cpp:2231]: Oh I see. Right.

**Comment by illwieckz** [comment]
I addressed the various comments.

**Comment by slipher** [comment]
> I also added VC4 detection, it's twice faster with float vertex on it.

Forgot to push something? I don't see any new stuff.

**Comment by illwieckz** [comment]
> > I also added VC4 detection, it's twice faster with float vertex on it.
> 
> Forgot to push something? I don't see any new stuff.

Indeed, now fixed.

**Comment by slipher** [comment]
LGTM

**Pull request merged**